### PR TITLE
Feature/wk web view

### PR DESCRIPTION
**Pull request by tjanela**
This PR leverages the previous #4430 from @Tripwire999

We have updated the httpbin.org certificate and its trust chain certificates.
We have updated the ```[AFWKWebView testProgressIsSet]``` test to use a large download URL.

### CODE REVIEW

**Comment by tjanela** [comment]
@kylef, @cnoon, @conradev, @SlaunchaMan,

It seems like the two last build configurations of travis are timing out.
Let me know where I can help.

**Comment by revolter** [comment]
Related to #4417.

**Comment by tjanela** [comment]
@revolter, most of the tests are failing because the SSL pinning code was using expired certificates. That is fixed in #4439, this PR. The only tests that failed did so due to timeout problems in the CI system as you can inspect here: 
https://travis-ci.org/AFNetworking/AFNetworking/jobs/585812060
I'm suspecting that the problem is related to a gem install failing:
```gem install fastlane --no-rdoc --no-ri --no-document --quiet```
But I have no clue how to replicate this problem. Maybe updating the CI yaml file?

**Comment by tjanela** [comment]
Inspecting the worker configuration of the build that failed versus the build that passed:
2980.10 uses macos10.12
2980.11, 2980.12 use osx10.11
Not sure if it is related.

**Comment by revolter** [comment]
> That is fixed in #4439, this PR.

I know, that's why I didn't say that it's duplicated; because yours is more complete. But it would have been nice for you to let me know that you did some similar work, so I get notified and know about it without finding it by mistake :D

> The only tests that failed did so due to timeout problems in the CI system

As for the hang issue, it sounds very very similar to https://travis-ci.community/t/build-times-out-with-no-apparent-reason/5083/4, and if I'm not mistaken, `fastlane` recently added the same `mini_magick` gem as a dependency.

**Comment by tjanela** [comment]
Got it @revolter, I'm sorry about that. Frankly, I didn't look further down on the open PR list. I did now and noticed you have some PRs that could really leverage some of the changes here (in fact any PR that triggers the CI will fail if the certificates are not updated).
Let me investigate the issue you are referencing. Maybe it shines some light.

**Comment by tjanela** [comment]
The build failed again due to the reasons indicated above.
@revolter I did check your reference and things are somewhat different between the builds.
The commonality is fastlane, for sure. But in your reference the process seems to fail when using fastlane and in AFNetworking builds things fail installing fastlane when the os is osx10.11
I opened a community topic:
https://travis-ci.community/t/build-stalling-on-gem-install-fastlane/5221
Let's see if someone has some pointers.
What I'll do next is find the immediate previous build that worked and forcefully install the fastlane version used.

**Comment by revolter** [comment]
So it doesn't work with fastlane 2.109.1, right?

**Comment by tjanela** [comment]
@revolter it **works** with 2.109.1
I chose this particular version by navigating to the last build that worked on osx10.11 and getting the fastlane version from the build log.

I updated the travis config to use that particular version and the gem install fastlane started working as expected.

There seems to be a failing test related to ios 9.3 I'm tracking down and that is why the build is failing now.

**Comment by revolter** [comment]
I see. Could you try with `2.127.1`?

**Comment by revolter** [comment]
And then with `2.127.2`. It should still work with `.1`, but fail with `.2`.

**Comment by tjanela** [comment]
Trying with 2.127.1 now.

**Comment by tjanela** [comment]
@revolter Looks like 2.127.1 has conflicting dependencies. See:
https://travis-ci.org/AFNetworking/AFNetworking/jobs/590463758
Trying the latest version with verbose mode on.

**Comment by tjanela** [comment]
@kylef, @cnoon, @conradev, @SlaunchaMan,

This PR has been updated to pass all tests and fixes a problem with travis CI, osx10.11, and fastlane's recent versions.

Let us know if there is a timeline to address the UIWebView deprecation. This PR addresses it.

**Comment by revolter** [comment]
Good job @tjanela! 🎉

**Comment by tjanela** [comment]
@SlaunchaMan where can I help to get a new release with these fixes going?
Thanks.

**Comment by R01NDev** [comment]
Hello, is there an ETA for the merge, one of our projects depends on this and I need to give a time-frame. Thank you everyone for your hard work

**Comment by tjanela** [comment]
Hey, @R01NDev, I've contacted Alamofire directly, via email.
Ultimately they are the owners and maintainers of the AFNetworking project.
I've asked exactly those questions. Let's see what they come back with.
I'll keep you posted.

**Comment by tjanela** [comment]
@kylef, @cnoon, @conradev, @SlaunchaMan,
The community needs you :)
Let us know if this PR has a chance of being reviewed, and more importantly, when.

**Comment by R01NDev** [comment]
@tjanela Hello, sorry for replying here again but I'm being pressured by management to get things going, is there any other solution than just sitting around waiting? My best regards to you

**Comment by revolter** [comment]
You can fork this repo, manually merge this PR in your fork, and then use it in the Podfile.

**Comment by tjanela** [comment]
> @tjanela Hello, sorry for replying here again but I'm being pressured by management to get things going, is there any other solution than just sitting around waiting? My best regards to you

@R01NDev @revolter's approach is the indicated approach and the one we are close to using in our dependencies. Bear in mind that this PR should be carefully reviewed by your team. If you find anything let us know and we will collaborate on a fix.

Naturally, we would prefer to get this approved and out via the official channels. I trust that the maintainers will eventually get to this, it's just a matter of waiting.

**Comment by tjanela** [comment]
@kylef, @cnoon, @conradev, @SlaunchaMan,

This PR addresses changes that have occurred in the App Store Approval process. These changes are **causing rejection for new apps** if they reference UIWebView from UIKit.
As such, it is of utmost importance that we get something that addresses this out to the community.
This PR covers not only those issues, leveraging previous work from @Tripwire999, but also addresses some timeout errors that were occurring on Travis CI.
It bears without saying that the community extensively uses this repository, and we would appreciate a signal from you.

Thank you for your time in advance.

**Comment by jshier** [comment]
@tjanela Thanks for the hard work here! Unfortunately, we can't mere this all at once, as it's a rather breaking change. If you have time, please separate the CI fixes from the `UIWebView` removal and I can get those merged immediately for a 3.4 release. Then, once that's released we can merge the breaking `UIWebView` changes for version 4.

As an aside, no new apps should be using this library. Please, please, transition to Alamofire, one of the other networking libraries, or roll your own `URLSession` wrapper.

**Comment by tjanela** [comment]
@jshier, of course. Thank you for taking the time to review the open PRs.
I will do what you require in the coming days.
As instructed, we will create a PR that contains the CI fixes, and then create a new one, leveraging those fixes, with the UIWebView removal.

We do not use AFNetworking directly on any new project. The case at hand, for us specifically, is a dependency that relies on AFNetworking which we need updated. They have no roadmap to cease AFNetworking dependency.

**Comment by tjanela** [comment]
@jshier, this is the third, and last, in a series of PRs that aim to breakdown #4439 into smaller chunks.
This one contains UIWebView removal, and only the removal, favoring WKWebView instead.

**Comment by Gordon2780** [comment]
> @jshier, this is the third, and last, in a series of PRs that aim to breakdown #4439 into smaller chunks.
> This one contains UIWebView removal, and only the removal, favoring WKWebView instead.

Can you tell me how to point 'pod' to "4439 PR"? When I use the pod command "pod 'afnetworking', GIT: 'https://github.com/afnetworking/afnetworking.git',: Branch = > feature / wkwebview '" "pod update",  will prompt "fatal: remote branch feature / wkwebview not found in upstream origin" Thank you

**Comment by bofeizhu** [comment]
@jshier @tjanela Could we merge #4472 first and release 3.4?

**Comment by hyunwoo3008** [comment]
How much longer do we have to wait

**Comment by davide1945** [comment]
@jshier, @tjanela, @kylef, @cnoon, @SlaunchaMan
Any news about the release of AFNetworking without UIWebView? Thank you very much in advance

File: UIKit+AFNetworking/WKWebView+AFNetworking.m [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// UIWebView+AFNetworking.m
+// WkWebView+AFNetworking.m
```
Comment (flybrotherlixiang) [inline UIKit+AFNetworking/WKWebView+AFNetworking.m]: 123

**Comment by prabh08** [comment]
I am also facing same issue. Any update?

**Comment by rogerluan** [comment]
Any update on this?

**Comment by bofeizhu** [comment]
@kylef, @cnoon, @conradev, @SlaunchaMan @jshier 

https://developer.apple.com/news/?id=12232019b

> Updating Apps that Use Web Views
December 23, 2019
If your app still embeds web content using the deprecated UIWebView API, we strongly encourage you to update to WKWebView as soon as possible for improved security and reliability. WKWebView ensures that compromised web content doesn’t affect the rest of an app by limiting web processing to the app’s web view. And it’s supported in iOS and macOS, and by Mac Catalyst.
The App Store will no longer accept new apps using UIWebView as of April 2020 and app updates using UIWebView as of December 2020.

Could we move forward with this PR please?

**Comment by pramodsharma403** [comment]
Any expected date of new version release?

**Comment by Bintong** [comment]
Good job @tjanela! 🎉

**Comment by dmitry-kovba** [comment]
Any updates?

**Comment by rogerluan** [comment]
Anyone got any updates? 😄

**Comment by jshier** [comment]
We'll be getting a 3.3 update out soon, with bug fixes and updates for the latest OSes. Then we'll ship 4.0, which removes `UIWebView` and replaces it with `WKWebView`. Apple won't stop accepting new apps using `UIWebView` until April 2020, and won't stop accepting updates to apps using it until December 2020. See their press release [here](https://developer.apple.com/news/?id=12232019b). There are no concrete dates for AFNetworking releases, other than to say we'll get them out before the hard deadlines hit.

In the meantime, if you're not using the `UIWebView` extensions, you can avoid the issue entirely by using our CocoaPods subspaces to pull in only the components you want, or by removing the `UIWebView` extension from your AFNetworking checkout, such as through Carthage.

**Comment by tjanela** [comment]
Thanks for the update @jshier. Let me know if these conflicts should be solved and I'll find the time to deal with them.

**Comment by jshier** [comment]
Once 3.3 is out would probably be the best time for that, but you can fix them in the meantime if you want.

**Comment by tjanela** [comment]
I assume a merge from master would be the correct way to go about this. I'll do it in the coming week if time allows.

**Comment by rogerluan** [comment]
Thank you so much for your update @jshier ! We'll try to use the subspaces to avoid using web view. Thanks! 👍

**Comment by steventsng** [comment]
Hello, do we have any idea of when this would be merged? Just got a email from Apple that they would be stop accepting submissions in April 2020.

**Comment by 3a4oT** [comment]
@jshier ping.

**Comment by harrisrap** [comment]
@jshier Any updates on this? April is coming quick, and we need to get this fix in our apps ASAP. Can you at least give us an estimate on when you think it will be available? Thanks.

**Comment by SurendraK11** [comment]
Hi @tjanela , any further update for this PR...
Soon I'm looking forward to submit my app for review, I do not want to push app with this warning... Please resolve this issue

**Comment by SurendraK11** [comment]
Hi @tjanela, please unblock us by merging this PR and create new build, so that we can push our apps for apple review

**Comment by tjanela** [comment]
@SurendraK11, you are more than welcome to point your podfile directly at our repo. That should unblock you.

As for release dates of AFNetworking, please see the maintainers’ comment:
https://github.com/AFNetworking/AFNetworking/pull/4439#issuecomment-574752353

**Comment by jshier** [comment]
Apparently UIWebView doesn't compile on Catalyst at all, so I think we'll have to take this change now and jump right to 4.0.

**Comment by internalG** [comment]
Is this commit to guarantee that AFNetworking and WKWebView share the same session cookie automatically?

**Comment by ElfSundae** [comment]
`loadData:MIMEType:characterEncodingName:baseURL:` requires iOS 9+

**Comment by tjanela** [comment]
Thanks, @jshier!

**Comment by jshier** [comment]
> `loadData:MIMEType:characterEncodingName:baseURL:` requires iOS 9+

Good point @ElfSundae, guess 4.0 will require iOS 9.

**Pull request merged**